ui/src/components/ChoiceModal.tsx

```diff
@@ -0,0 +1,93 @@
+
+import React from 'react'
+import { Choice, ContractId } from '@daml/types';
+import { useLedger } from '@daml/react';
+
+import { Formik, Form, Field, FieldProps, FormikHelpers, FieldAttributes, useField } from 'formik';
+import { Share } from 'phosphor-react';
+import Select, { Props } from 'react-select';
+import Modal from './Modal';
+import DayPicker from "./DayPicker";
+
+export const Nothing = Symbol('Nothing');
+type Nothing = typeof Nothing;
+
+type PartialMaybe<T> = {
```

Comment:
Each field is made optional without edge cases, provided nothing else uses `Nothing`.